src/mds/MDSMap.cc

```diff
@@ -462,6 +459,9 @@ void MDSMap::mds_info_t::decode(bufferlist::iterator& bl)
   if (struct_v >= 6) {
     ::decode(standby_for_fscid, bl);
   }
+  if (struct_v >= 7) {
+    ::decode(standby_replay, bl);
+  }
```

Comment:
In the case of a daemon that is not in standby replay state but wants to be: old MDS daemons will continue to send beacons that request state standby replay, which MMDSBeacon::decode will interpret as standby_replay=true, which gets picked up in MDSMonitor::prepare_beacon.